distributed-lock-core/src/main/java/com/github/alturkovic/lock/configuration/DistributedLockConfiguration.java

```diff
@@ -40,15 +44,27 @@
 @Configuration
 @EnableAspectJAutoProxy
 @ComponentScan("com.github.alturkovic.lock.converter")
+@Slf4j
 public class DistributedLockConfiguration {
 
   @Bean
   public LockAdvice lockAdvice(final IntervalConverter intervalConverter, final KeyGenerator spelKeyGenerator, final List<Lock> locks) {
-    return new LockAdvice(intervalConverter, spelKeyGenerator, locks.stream().collect(Collectors.toMap(Lock::getClass, Function.identity())));
+    return new LockAdvice(intervalConverter, spelKeyGenerator, locks.stream().map(l -> {
+      if (AopUtils.isAopProxy(l) && l instanceof Advised) {
+        final Advised advised = (Advised) l;
+        try {
+          return (Lock) advised.getTargetSource().getTarget();
+        } catch (Exception e) {
+          log.error("Can't get lock type from AOP Proxy");
```

Comment:
Please include exception in error log

Reply:
Upon further consideration, I believe it would be better if the configuration failed fast. Please throw a 'DistributedLockException' with the same message and pass the exception to it. If however, you feel that logging is a better choice, please explain, since I don't have experience with CGLib. Thank you!

Reply:
I think it is ok to throw exception, maybe it is best choice log and throw exception?